Add backup/restore plans (#339)
puppetlabs/puppetlabs-peadm: documentation/backup_restore.md

   ```
    bolt plan run peadm::util::init_db_server db_host=my.primary_db.vm pe_version=2023.5.0 install_pe=true
   ```

    This plan will perform the following:

     1. Clean the current certificate for the database server from the primary server.
     1. Request a new certificate for the database server with the right extensions (peadm_role = puppet/puppetdb-database, peadm_availability_group=A).
     1. Stop the puppetdb service on the compilers.
     1. Prepare a `pe.conf` file on the database server for database installation
     1. Install PE on the database server using the generated `pe.conf` file.
     1. Configure the database as the primary puppetdb database in the XL installation.
     1. Run puppet on the compilers to allow puppetdb on the compilers to be reconfigured with the new primary database server.
     1. Start the puppetdb service on the compilers.
     1. Restart the puppetserver service on the compilers.

1. Perform a `recovery-db` restore of your primary database server using the backup. For example:
   ```
    bolt plan run peadm::restore --targets my.primary.vm input_file="/tmp/my_backup.tar.gz" restore_type=recovery-db
   ```

>    **Note**: You should use the `restore_type=recovery-db` parameter to recover the database server. 

Reviewer: ```suggestion
   **Important**: You must use the `restore_type=recovery-db` parameter to recover the database server. 
```
